packages/server/src/automations/steps/branch.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  AutomationActionStepId,
+  AutomationCustomIOType,
+  AutomationIOType,
+  AutomationStepDefinition,
+  AutomationStepType,
+} from "@budibase/types"
+
+export const definition: AutomationStepDefinition = {
+  name: "Branch",
+  icon: "Branch3",
+  tagline: "Branch from this step",
+  description: "Branching",
+  stepId: AutomationActionStepId.BRANCH,
+  internal: true,
+  features: {},
+  inputs: {},
+  schema: {
+    inputs: {
+      properties: {
+        branches: {
+          properties: {
+            name: {
+              type: AutomationIOType.STRING,
+            },
+            condition: {
+              customType: AutomationCustomIOType.FILTERS,
+            },
+          },
+        },
+        children: {
+          type: AutomationIOType.ARRAY,
+        },
+      },
+      required: ["conditions"],
+    },
+    outputs: {
+      properties: {
+        output: {
```

Comment:
I think the branch step should output which branch it decided to go down, so the branch name and the result of the condition, similar to the Filter step today - I don't think there is must else it can output.